app/views/admin/api_docs/base/swagger.html.erb

```diff
@@ -37,18 +37,80 @@
       }
     }
 
+    var AUTH_METHOD_USER_KEY = 1;
+    var AUTH_METHOD_ID_KEY_PAIR = 2;
+
+    function getAuthParamsForMethod(authMethod) {
```

Comment:
There could be less repetition here, for example like this (there are more options):
```js
function getAuthParamsForMethod(authMethod) {
  if(authMethod !== AUTH_METHOD_ID_KEY_PAIR && authMethod !== AUTH_METHOD_USER_KEY) { return []; }
  
  var authParam = {
    name: '<%= @api_docs_service.service.proxy.auth_app_key %>',
    in: 'header',
    'x-data-threescale-name': 'app_keys',
    required: true,
    type: 'string'
  };
  
  if(authMethod === AUTH_METHOD_USER_KEY) { return [Object.assign(authParam, {description: 'User Key authentication parameter'})] }
  return [
    Object.assign(authParam, {description: 'App Key authentication parameter'}),
    {
      name: '<%= @api_docs_service.service.proxy.auth_app_id %>',
      in: 'header',
      description: 'App ID authentication parameter',
      'x-data-threescale-name': 'app_ids',
      required: true,
      type: 'string'
    }
  ];
}
```